homeassistant/components/sensor/mold_indicator.py

```diff
@@ -0,0 +1,255 @@
+        outdoor_temp_sensor, indoor_humidity_sensor,
+        calib_value)])
+
+
+# pylint: disable=too-many-instance-attributes
+class MoldIndicator(Entity):
+    """Represents a MoldIndication sensor."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, name, indoor_temp_sensor, outdoor_temp_sensor,
+                 indoor_humidity_sensor, calib_value):
+        """Initialize the sensor."""
+        self._state = "-"
+        self._hass = hass
+        self._name = name
+        self._indoor_temp_sensor = indoor_temp_sensor
+        self._indoor_humidity_sensor = indoor_humidity_sensor
+        self._outdoor_temp_sensor = outdoor_temp_sensor
+        self._calib_value = calib_value
+        self._is_metric = (hass.config.temperature_unit == TEMP_CELCIUS)
+
+        self._dewpoint = None
+        self._indoor_temp = None
+        self._outdoor_temp = None
+        self._indoor_hum = None
+        self._crit_temp = None
+
+        track_state_change(hass, indoor_temp_sensor, self._sensor_changed)
+        track_state_change(hass, outdoor_temp_sensor, self._sensor_changed)
+        track_state_change(hass, indoor_humidity_sensor, self._sensor_changed)
+
+        # Read initial state
+        self._sensor_changed(indoor_temp_sensor,
+                             None, hass.states.get(indoor_temp_sensor))
+        self._sensor_changed(outdoor_temp_sensor,
+                             None, hass.states.get(outdoor_temp_sensor))
+        self._sensor_changed(indoor_humidity_sensor,
+                             None, hass.states.get(indoor_humidity_sensor))
+
+    @staticmethod
+    def _update_temp_sensor(state):
+        """Parse temperature sensor value."""
+        unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        temp = util.convert(state.state, float)
+
+        if temp is None:
+            _LOGGER.error('Unable to parse sensor temperature: %s',
+                          state.state)
+            return None
+
+        # convert to celsius if necessary
+        if unit == TEMP_FAHRENHEIT:
+            return util.temperature.fahrenheit_to_celcius(temp)
+        elif unit == TEMP_CELCIUS:
+            return temp
+        else:
+            _LOGGER.error("Temp sensor has unsupported unit: %s"
+                          " (allowed: %s, %s)",
+                          unit, TEMP_CELCIUS, TEMP_FAHRENHEIT)
+
+        return None
+
+    @staticmethod
+    def _update_hum_sensor(state):
+        """Parse humidity sensor value."""
+        unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        hum = util.convert(state.state, float)
+
+        if hum is None:
+            _LOGGER.error('Unable to parse sensor humidity: %s',
+                          state.state)
+            return None
+
+        # check unit
+        if unit != "%":
+            _LOGGER.error(
+                "Humidity sensor has unsupported unit: %s %s",
+                unit,
+                " (allowed: %)")
```

Comment:
Why not `_LOGGER.error("Humidity sensor has unsupported unit: %s (allowed: %)", unit)`? There is not much advantage for using two vars.